app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +280,141 @@ angular.module('confRegistrationWebApp')
     $scope.hasPendingCheckPayment = function(payments){
       return _.some(payments, { paymentType: 'CHECK', status: 'PENDING' });
     };
+
+    ////// START EVENT-433 //////
+
+    // Get spouse registration
+    // End point: getSpouseRegistration = /conferences/conferenceId/registrations/spouse
+    ///// TEST DATA change end point url below to: /conferences/conferenceId/registrations/current /////
+    $http.get('conferences/' + conference.id + '/registrations/spouse')
+      .then(function(response) {
+        if (response.status === 200) {
+          $scope.spouseRegistration = response.data;
+
+          ///// TEST DATA /////
+          // $scope.spouseRegistration.completed = true;
+          // $scope.spouseRegistration.registrants[0].email = 'janeworner3@hotmail.com';
+          // $scope.spouseRegistration.registrants[0].firstName = 'Jane';
+          // $scope.spouseRegistration.registrants[0].lastName = 'Worner3';
+          ///// TEST DATA ////
+
+        }
+        else {
+          $scope.spouseRegistration = {};
+          console.log('No current and/or spouse registration found for user with ID.');
+        }
+      }).catch(function(response) {
+      $scope.spouseRegistration = {};
+      console.log('No current and/or spouse registration found for user with ID. Status = ' + response.status);
+      console.log(response.data);
+    })
+      .finally(function() {
+        checkSpouseRegistration();
+      });
+
+    //Parameters needed to retrieve all current registrants for conference.id
+    var queryParameters = {
+      block: [],
+      //page: 1,
+      //limit: 20,
+      orderBy: 'last_name',
+      order: 'ASC',
+      filter: '',
+      filterPayment: '',
+      filterRegType: '',
+      includeCheckedin: 'yes',
+      includeWithdrawn: 'yes',
+      includeIncomplete: 'yes'
+    };
+
+    //Get all current registrants based on conference.id and queryParameters
+    //then check if spouse is registered and display 'Spouse Registration' section if needed.
+    var checkSpouseRegistration = function() { // jshint ignore:line
+      $http.get('conferences/' + conference.id + '/registrations', {params: queryParameters})
```

Comment:
In the original longer version of the code that I had I was using it but we dropped it when shortening the code. We actually need this HTTP call and the response because the response will contain a complete list of all registrants that have already registered for the event that we can then use to check against spouseRegistration.  currentRegistration does not contain this full list. Only the existing current registrant. So I'm glad you saw this.  
I resplaced:
_var spouseIncludedOnCurrentRegistration = _.contains(_.map($scope.spouseRegistration.registrants, 'email'), $scope.currentRegistration.registrants[0].email);_
With this:
_var spouseIncludedOnCurrentRegistration = !_.isEmpty(_.intersection(_.map($scope.spouseRegistration.registrants, 'email'),
            _.map(_.map(_.map(response.data.registrations, 'registrants'), 0), 'email')));_

On a different topic.  I wasn't able to find a better solution to showing/hiding the button than using _sessionStorage.setItem('spouseSectionVisible', true)_ and _sessionStorage.getItem('spouseSectionVisible')_.
If you have a better suggestion let me know.

Also, I noticed in testing that we are deleting the registration to early.
So that when the page refreshes it will not display the new registration.
I tried using 'finally' in the HTTP call chain we are using but that didn't help.
If you have a suggestion let me know.

Finally, I'm not positive but I think the payload examples that you and Jon came up may not be correct.  Only because when it does add the new registration to the event it is using the same duplicate registrant twice.  The previous code I had showed both old and new registrant.  But I'm not sure we will be able to verify this until we have test data to test against.